crates/bevy_pbr/src/material.rs

```diff
@@ -180,6 +180,13 @@ pub trait Material: Asset + AsBindGroup + Clone + Sized {
     ) -> Result<(), SpecializedMeshPipelineError> {
         Ok(())
     }
+
+    /// Specify a batch key to use for shadows. Returning Some(value) allows batching of distinct materials which will produce
+    /// the same depth-prepass output (such as opaque [`StandardMaterial`]s with different textures), rendering with the first
+    /// material of the batch for all meshes.
+    fn shadow_material_key(&self) -> Option<u64> {
```

Comment:
This feels pretty hacky :/. I'd rather we just have a separate set of bindings for shadow passes, or split bindings between vertex/fragment.

Reply:
I considered how to do this without requiring large user changes, I couldn’t see a way. 

Ideally you want this to work for standard materials without extra user input, so 
- a separate component (`ShadowMaterial(Handle<M>)`) doesn't really work unless we also add a system to add the component automatically where appropriate, and that system would need to do lookups into the assets resource to make the determination, which sounds expensive and still messy. 
- have the key function return a handle/asset id, this requires either adding the handle(s) to every material or using some resource to store them and giving access in the key function. This gets at least as awkward, particularly if you have more than one groupable set (I have this case in my work code where some opaque materials can be grouped but not all, based on the “plot” they belong to in the world). Also adds effort around extracting/preparing the new handles, basically seems like a lot of extra complexity for limited reward. 

If you can see a clean way to do this I’d be happy to change it, but I don’t see it right now.

Reply:
I suppose for now this is the best we can do. I think for 0.14 we might want to look into redoing the Material API so we can support batching, bindless, etc easier.